index.js

```diff
@@ -138,3 +138,76 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+// EVENT DELEGATION
+// Identify which elements you care to listen for an event on
+//      'clicks'
+
+// delete and likes buttons of all the posts
+// Find the closest common parent of all these elements
+//      <ul id="animals"> == $0
+
+
+//can click on 'like' 'delete' and 'toggle update form' for each animal sighting post
+const animalsCollection = document.querySelector('ul#animals')
+
+animalsCollection.addEventListener('click', function (event) {
+
+    const li = event.target.closest('li')
+    
+    // before review
+    // if (event.target.matches('button.like-button')){
+    //     const likesPtag = event.target.previousElementSibling
+    //     const likes = parseInt(likesPtag.textContent) + 1
+    //     likesPtag.textContent = `${likes} Likes`
+    // }
+
+    //after review
+    if (event.target.matches('button.like-button')){
+        const likesPtag = li.querySelector('p.likes-display')
+        const currlikes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${currlikes} Likes`
+    }
+
+    //before review
+    // else if (event.target.matches('button.delete-button')) {
+    //     const animals = event.target.closest('li')
+    //     animals.remove()
+    // }
+    // getting error: index.js:163 Uncaught TypeError: Cannot read property 'remove' of null
+    // at HTMLUListElement.<anonymous> (index.js:163)
+    // when trying to click on 'delete' button. not sure what is wrong with the code.
+
+    //after review
+    else if(event.target.className === 'delete-button') {
+        li.remove()
+    }
+    else if(event.target.matches('button.toggle-update-form-button')) {
+        
+
+         const form = li.querySelector('form')
+
+        // if(form.style.display === 'block') {
+        //     form.style.display
+        // }
+        // else {
+        //     form.style.display = 'block'
+        // }
+
+        form.style.display = form.style.display === 'block' ? 'none' : 'block'
+
+    }
+    
+})
```

Comment:
Nice work. I like your variable name for the animals UL. I like your approach. You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')
animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```